### PR TITLE
Added workflow for creating and pushing git tag (update actions/checkout to the version 4)

### PR DESCRIPTION
**Pull request by Nikita-Smirnov-Exactpro**

### CODE REVIEW

File: .github/workflows/compaund-git-tag-push.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    steps:
+      - uses: actions/checkout@v3
+      - name: Show tag name
+        run: |
+          echo "tag ${{ inputs.tagName }}"
+          echo "description ${{ inputs.tagDescription }}"
+      - name: Create and push tag
+        run: |
+          git config --global user.name 'Github Actions'
+          git config --global user.email 'exactpro@exactpro.com'
```
Comment (OptimumCode) [inline .github/workflows/compaund-git-tag-push.yml:30]: This is usually the email that is used for auto-commits in GitHub:
name = 'github-actions'
email = '41898282+github-actions[bot]@users.noreply.github.com'

Probably, we should use this one as well

**Review by OptimumCode — APPROVED** [review]

**Review by OptimumCode — APPROVED** [review]
Just a small note: the PR is related to adding a new create-tag step. But we also updated the dependencies. It is not a good practice to mix changes...

**Comment by Nikita-Smirnov-Exactpro** [comment]
> Just a small note: the PR is related to adding a new create-tag step. But we also updated the dependencies. It is not a good practice to mix changes...

I noticed the warning related to https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/
and added updates into this change.
Mix changes is a bad practice because revert operation will be more difficult, also, we apply not declared change?

**Comment by OptimumCode** [comment]
> Mix changes is a bad practice because revert operation will be more difficult, also, we apply not declared change?

If this is the question about why mixing changes is a bad practice then yes and yes - the original purpose of the PR is to add workflow for adding tags. And it should contain only changes that correspond to that. If we need to update dependencies or etc we can open another PR...
Mixing changes in a single PR also does not help automated workflows, e.g. automatically collecting release notes

**Comment by Nikita-Smirnov-Exactpro** [comment]
> Mixing changes in a single PR also does not help automated workflows, e.g. automatically collecting release notes
It is a good point, thank you )

**Pull request merged**